lib/dor/services/client.rb

```diff
@@ -27,7 +27,12 @@ class Client
       class Error < StandardError; end
 
       # Error that is raised when the ultimate remote server returns a 404 Not Found for the id in our request (e.g. for druid, barcode, catkey)
-      class NotFoundResponse < Error; end
+      class NotFoundResponse < Error
+        def initialize(response, object_identifier)
+          @json = response.body
```

Comment:
do you need/want an `attr_accessor` for `@json` to keep consistent with the `#json` method on `UnexpectedResponse` and its subclasses?